errors.go

```diff
@@ -23,389 +23,136 @@
 // is of a certain type.
 package errdefs
 
-import (
-	"context"
-	"errors"
+import "context"
+
+type code int
```

Comment:
This could use an int8.

This type could also just be the base error type without wrapping. Then keep the error struct only when adding a custom message.

Reply:
I restored the `customMessage` type and the wrapping code related to that and changed this type to an `int8` and renamed this type to `Error`. This will still have the consolidated type but it should be more simple.